core/array/shared/slice.rb

```diff
@@ -533,6 +533,25 @@ def to.to_int() -2 end
     a.send(@method, eval("(-9...)")).should == nil
   end
 
+  ruby_version_is "3.0" do
+    it "can be sliced with Enumerator::ArithmeticSequence" do
+      a = [0, 1, 2, 3, 4, 5]
+      a.send(@method, eval("(0..).step(2)")).should == [0, 2, 4]
```

Comment:
```suggestion
      a.send(@method, eval("(0..).step(1)")).should == [0, 1, 2, 3, 4, 5]
      a.send(@method, eval("(0..).step(2)")).should == [0, 2, 4]
```
to add a basic example with step 1